npm/private/npm_link_package.bzl

```diff
@@ -208,18 +229,25 @@ def _impl_store(ctx):
 deps of npm_link_package_store must be in the same package.""" % (ctx.label.package, dep[_StoreInfo].root_package)
                 fail(msg)
             dep_package = dep[_StoreInfo].package
-            dep_version = dep[_StoreInfo].version
             dep_aliases = _dep_aliases.split(",") if _dep_aliases else [dep_package]
             dep_virtual_store_directory = dep[_StoreInfo].virtual_store_directory
             if dep_virtual_store_directory:
                 for dep_alias in dep_aliases:
                     # "node_modules/{virtual_store_root}/{virtual_store_name}/node_modules/{package}"
                     dep_symlink_path = paths.join("node_modules", utils.virtual_store_root, virtual_store_name, "node_modules", dep_alias)
-                    dep_symlink = ctx.actions.declare_file(dep_symlink_path)
-                    ctx.actions.symlink(
-                        output = dep_symlink,
-                        target_file = dep_virtual_store_directory,
-                    )
+                    if ctx.attr.allow_unresolved_symlinks:
+                        dep_symlink = ctx.actions.declare_symlink(dep_symlink_path)
+                        ctx.actions.symlink(
+                            output = dep_symlink,
+                            target_path = relative_file(dep_virtual_store_directory.path, dep_symlink.path),
```

Comment:
Why does it have to be relative in this case? Can it not also be relative in the other?

Reply:
`ctx.actions.symlink` takes a File and not a path and it decides on the symlink path itself. Bazel lays out symlinks with absolute paths with that style.